### DISCUSSION TITLE
endpoint error

### DISCUSSION THREAD

**Discussion by rick0403** [saleweaver/python-amazon-sp-api#47] [General]
Hi All: Just trying to get the API to work for the first time. Trying to retrieve orders using only the basic default code. Getting an error AttributeError: 'tuple' object has no attribute 'endpoint' .  I guess I am missing something in the specification of the marketplace attribute?  
I'm new to AWS. I found  this guide on service endpoints  https://docs.aws.amazon.com/general/latest/gr/rande.html
and I see this "protocol://service-code.region-code.amazonaws.com"
which looks similar to the URL string in the marketplace attribute, but I can't find a permutation that works
Please help Thank you!

My code:
```
from sp_api.api import Orders
from sp_api.base import SellingApiException
from datetime import datetime, timedelta 

credentials = dict(...)

try:
    result = Orders(marketplace=('https://sellingpartnerapi-na.amazon.com', 'ATVPDKIKX0DER', 'us-east-1'), credentials=credentials).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())
    print(result.payload)
except SellingApiException as ex:
    print(ex)
```
Here's the error msg in full:
 ```
Traceback (most recent call last):
  File "C:\Users\rdmos\Desktop\getAmazonOrders.py", line 25, in <module>
    result = Orders(marketplace=('https://sellingpartnerapi-na.amazon.com', 'ATVPDKIKX0DER', 'us-east-2'), credentials=credentials).get_orders(CreatedAfter=(datetime.utcnow() - timedelta(days=7)).isoformat())
  File "C:\Users\rdmos\Miniconda3\envs\ball\lib\site-packages\sp_api\base\client.py", line 38, in __init__
    self.endpoint = marketplace.endpoint
AttributeError: 'tuple' object has no attribute 'endpoint'
```

**Comment by saleweaver** [comment]
You have to import Marketplaces and use 

    Marketplaces.UK

for the uk marketplace.

**Reply by rick0403** [reply]
Ah ok. Thanks!  I dropped when copying from someone else's code.  I am US, so that would be Marketplaces.US

**Reply by saleweaver** [reply]
For US you can leave marketplace out completely, that's the default.

**Reply by rick0403** [reply]
It's working! Thank you for your help.. 
My last issue was: There is a key in the credentials called "lwa_app_id"  which doesn't map exactly to an identical name somewhere else.  In Developer Central, there is an "App ID" (don't use this) but also under **LWA Credentials** there is a "Client Identifier" (use this).
